xconn/client.py

```diff
@@ -13,16 +15,23 @@ def __init__(
         authenticator: auth.IClientAuthenticator = AnonymousAuthenticator(""),
         serializer: serializers.Serializer = serializers.JSONSerializer(),
         ws_config: types.WebsocketConfig = types.WebsocketConfig(),
+        on_connect: Callable[[Any], Any] | None = None,
```

Comment:
Client(s) should have method "add_on_connect_listener" and "add_on_disconnect_listener" that takes a callback as argument. It should be possible to add multiple listeners for each time, so you should maintain a list internally.